subprojects/model-core/src/main/java/org/gradle/internal/instantiation/generator/AsmBackedClassGenerator.java

```diff
@@ -669,8 +690,28 @@ private void visitDeclaredAnnotationsOf(Constructor<?> constructor, MethodVisito
             }
         }
 
+        /**
+         * Generates the init method with deprecation logging disabled.
+         *
+         * This is necessary because the initialization work invokes property getters on the decorated instance.
```

Comment:
what's an example of this?

Reply:
An extension with a deprecated property with the getter that emits deprecation warnings. We have some in the codebase, e.g. `KotlinPluginOptions.jvmTarget`.